server/src/main/java/org/elasticsearch/cluster/metadata/MetadataDataStreamsService.java

```diff
@@ -133,6 +138,46 @@ public Tuple<ClusterState, ClusterStateAckListener> executeTask(
             }
         };
         this.updateOptionsTaskQueue = clusterService.createTaskQueue("modify-data-stream-options", Priority.NORMAL, updateOptionsExecutor);
+        ClusterStateTaskExecutor<UpdateSettingsTask> updateSettingsExecutor = new SimpleBatchedAckListenerTaskExecutor<>() {
+
+            @Override
+            public Tuple<ClusterState, ClusterStateAckListener> executeTask(
+                UpdateSettingsTask updateSettingsTask,
+                ClusterState clusterState
+            ) throws Exception {
+
+                ProjectMetadata projectMetadata = clusterState.projectState(updateSettingsTask.projectId).metadata();
```

Comment:
No need to convert to `ProjectState` here
```suggestion
                ProjectMetadata projectMetadata = clusterState.metadata().getProject(updateSettingsTask.projectId);
```

Reply:
I hadn't even realized that there were two ways to do it.